### PR TITLE
New Nav component

### PR DESCRIPTION
**Pull request by ShimiSun**
New Nav component that will render a collection of anchors.
Nav has an `items=[{label: 'my link', href='#'}...]` prop, but in case the user wants to use custom links or Buttons he can leverage the component's children instead. 

I moved the Sidebar example from Button to Nav, and refactored it to use Nav instead of a Box, I think it makes better sense for the example to be placed on Nav component.

Added on TS story example but tested all the other stories with TS as well.

### CODE REVIEW

File: src/js/components/Nav/Nav.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import { Anchor } from '../Anchor';
+import { Box } from '../Box';
+
+
+ // TODO resolve whether children should be allowed with items
```
Comment (ShimiSun) [inline src/js/components/Nav/Nav.js]: Opening this for discussion. I think there shouldn't be a problem to allow both.

File: src/js/components/Nav/Nav.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { Anchor } from '../Anchor';
+import { Box } from '../Box';
+
+// TODO resolve whether children should be allowed with items
+const Nav = ({ children, items = [], ...rest }) => (
+  <Box as="nav" flex={false} gap="medium" {...rest}>
+    {items.map(item => (
+      <Anchor href={item.href} label={item.label} key={item.label} />
```
Comment (ericsoderberghp) [inline src/js/components/Nav/Nav.js]: Why Anchor and not Button?

Comment (ericsoderberghp) [inline src/js/components/Nav/Nav.js]: How would the caller integrate this with a router?

Comment (ShimiSun) [reply src/js/components/Nav/Nav.js]: I found Anchor to be more simplified than Button. 
You can leverage the children to show a Button instead (or even a Box), see for example, the CustomChild story on storybook, so the option for alternative components is still there.

Comment (ShimiSun) [reply src/js/components/Nav/Nav.js]: Not sure why should she/he integrate with a router. 
I think the Nav should be separated from the routing logic, just like Anchor/Button doesn't know which router you are using, and just direct to the `href` provided. 
I might be missing something though, so feel free to elaborate.

Comment (ericsoderberghp) [reply src/js/components/Nav/Nav.js]: The routers have to tie in to the `onClick` property. So, you typically want to do something like `<RouterAwareAnchor path="/internal/path" label="Name" />` with the router taking care of `onClick` and `href`.

Comment (ShimiSun) [reply src/js/components/Nav/Nav.js]: I made the component less opinionated by removing `items`.

Comment (ShimiSun) [reply src/js/components/Nav/Nav.js]: I made the component less opinionated by removing `items`, I think it should solve the routing integration issue.

File: src/js/components/Nav/Nav.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import { Box } from '../Box';
+
+const Nav = ({ ...rest }) => (
+  <Box as="nav" flex={false} gap="medium" {...rest} />
```
Comment (ericsoderberghp) [inline src/js/components/Nav/Nav.js]: What's the rationale for `gap` defaulting to `medium`? I was envisioning using this in a Sidebar with Buttons, in which case I would want gap to be undefined.

File: src/js/components/Nav/stories/typescript/CustomAnchor.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      light: 'neutral-2',
+    },
+    hover: {
+      textDecoration: 'none',
+      fontWeight: 700,
+    },
+  },
+});
+
+const CustomAnchor = () => (
```
Comment (ericsoderberghp) [inline src/js/components/Nav/stories/typescript/CustomAnchor.tsx]: Can we remove this story, now that we don't have `items`?

File: src/js/components/Nav/Nav.js (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/Nav/Nav.js]: I agree it would have made more sense to use with `items`

File: src/js/components/Nav/stories/typescript/CustomAnchor.tsx (hunk 4 above)
Comment (ShimiSun) [reply src/js/components/Nav/stories/typescript/CustomAnchor.tsx]: done.

File: src/js/components/Nav/Nav.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/Nav/Nav.js]: So, are we going to remove the `gap="medium"` here?

File: src/js/components/Nav/stories/CustomChild.js [hunk 5]
```diff
@@ -30,10 +30,10 @@ const SidebarButtons = () => {
...
           ))}
-        </Box>
+        </Nav>
       </Box>
     </Grommet>
   );
 };
 
-storiesOf('Button', module).add('Sidebar', () => <SidebarButtons />);
+storiesOf('Nav', module).add('Custom child', () => <SidebarNav />);
```
Comment (ericsoderberghp) [inline src/js/components/Nav/stories/CustomChild.js]: Is it a "custom" child? Or is it just a normal use of Nav?

File: src/js/components/Nav/stories/CustomAnchor.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+const navItems = [
+  { label: 'HTML', href: '#' },
+  { label: 'JS', href: '#' },
+  { label: 'CSS', href: '#' },
+  { label: 'REACT', href: '#' },
+];
+
+const customTheme = deepMerge(grommet, {
+  anchor: {
```
Comment (ericsoderberghp) [inline src/js/components/Nav/stories/CustomAnchor.js]: This seems orthogonal to Nav, more about customizing Anchor.

File: src/js/components/Nav/stories/CustomChild.js (hunk 5 above)
Comment (ShimiSun) [reply src/js/components/Nav/stories/CustomChild.js]: I've replaced it with the name `Sidebar`, it doesn't make sense to have `Custom child` when we don't have `items`.

File: src/js/components/Nav/stories/CustomAnchor.js (hunk 6 above)
Comment (ShimiSun) [reply src/js/components/Nav/stories/CustomAnchor.js]: yet, I think it will be a helpful example. I can remove it if you strongly think it should be placed elsewhere. (I didn't find a Custom Anchor example on Anchor.)

**Comment by ShimiSun** [comment]
linking to #3580 #3489
